### ISSUE TITLE
Fixing indentation

### ISSUE THREAD

**Issue by jedmao** [slang800/editorconfig-tools#12]
What is your strategy for fixing indentation, especially converting from spaces to tabs. There is no way to assume a developer's intent between what was intended for indentation vs. what was intended for alignment. http://jedmao.ghost.io/2014/08/20/tabs-vs-spaces-the-age-old-war/

**Comment by slang800** [comment]
ref: https://github.com/slang800/editorconfig-tools/blob/d2c39482dc002528e543e4674e391a848c937b08/lib/rules/indent-char.coffee

Right now, we use a really naïve method: we detect the indentation on a full-file basis using [detect-indent](https://github.com/sindresorhus/detect-indent), then just replace every leading instance of the detected indent char with the new indent char. We don't make any attempt to determine what is alignment and what isn't, or how the new indentation would change the alignment, but if you have a good algorithm for that, I'd love to hear it.

Oh, and personally, I don't align and am trying to implement a very different solution to the alignment problem: https://github.com/slang800/slang.cx/blob/master/content/aligning-operators.md. So, I'm not super motivated to work on detecting/fixing mixed indentation and alignment in this tool. I'd love to take a PR for it though!

**Comment by jedmao** [comment]
My point is that it can't be done w/o a concrete syntax tree. There is no algorithm. It is, literally, impossible.

Any detection on indentation will emit false errors where spaces are used for leading alignment. The most that could be done is a warning saying that you shouldn't have spaces for leading alignment, but that's not really up to EditorConfig to decide, so that's not right either.

**Comment by slang800** [comment]
I know that it is impossible to do perfectly, but I'm sure there are ways to get closer than what we currently have.

And what would you propose? It would be easy to just add a note about this to the readme saying "if you use alignment and we detect that indentation needs to be fixed, then you may need to manually re-align"... but is there something better?

**Comment by jedmao** [comment]
There's really not.

**Comment by jedmao** [comment]
I discussed the matter in depth with my wife and came to the conclusion that fixing `indent_size` just shouldn't even be attempted. Fixing `indent_style`, however, is much more manageable and I'm doing that with [eclint](https://github.com/jedmao/eclint/blob/master/lib/rules/indent_style.ts#L95-L122). Be curious to know what your thoughts and opinions are on the matter and especially the technique with which I fixed `indent_style`. Always open for suggestions.
